### ISSUE TITLE
How to create own Nuget v3 feed?

### ISSUE THREAD

**Issue by maartenba** [NuGet/NuGetGallery#2844]
_From @SharpNoiZy on November 24, 2015 6:54_

Hi, maybe I'm stupid, blind or both :)
But how do I create my own NuGet feed with v3 like I done it for v2.
Install Nuget.Server into Webproject and it was done.
But this server is nuget v2.

Kind regards

_Copied from original issue: NuGet/Home#1756_

**Comment by maartenba** [comment]
_From @emgarten on November 24, 2015 19:31_

There's no NuGet.Server v3 yet. To help with this would you tell me what your scenarios are? Will you have a v3 authenticated feed, and would you prefer that it be static content, how would you ideally want to host the v3 feed?

**Comment by maartenba** [comment]
_From @SharpNoiZy on November 24, 2015 20:21_

We have currently an private v2 feed only for our company.
I want to upgrade this to v3, to share also corefx nuget packages.

So, I need something running on one company reachable server, nothing local on my PC.

**Comment by maartenba** [comment]
_From @yishaigalatzer on November 25, 2015 1:58_

We do need to update the server, but you don't actually need a V3 server for that. @joyhui we need to update the nuget.server package.

Lets move the bug over to the nugetgallery.

[ ] we want to make sure that all the new packages are supported (DNXCORE50, generations, etc.)
[ ] use the new folder structure supported by nuget.exe init command

**Comment by maartenba** [comment]
_From @yishaigalatzer on December 31, 2015 1:27_

@maartenba is working on a new version of nuget.server, that should support your scenario.

@maartenba can you please verify that corefx packages can be hosted on the server, and can you share early bits with @SharpNoiZy

**Comment by maartenba** [comment]
@SharpNoiZy We don't have a v3 protocol server yet, but have `NuGet.Server` in the works with support for new target frameworks like mentioned above.

Can you try our new NuGet.Server package?
* `NuGet.Server` version `2.10.0-develop-0016`
* From feed: `https://www.myget.org/F/nugetbuild/api/v2`

Or in short:
`Install-Package NuGet.Server -Version 2.10.0-develop-0016 -Source https://www.myget.org/F/nugetbuild/api/v2 -IncludePrerelease`

**IMPORTANT** do make a backup of your current NuGet.Server's packages folder in case you are upgrading. Ideally install this new package into a new server and copy over packages.

Feedback on this new version is welcome.

**Comment by maartenba** [comment]
@SharpNoiZy Have you had the chance to play with these bits?

**Comment by SharpNoiZy** [comment]
Today I had the chance to test it.
The named version "2.10.0-develop-0016" didn't work, but your latest beta version "2.10.0-ft-2843-0019" does the job! :+1: 

Kind regards

**Comment by maartenba** [comment]
Sweet, thanks! Did the " develop "  version throw any errors?

**Comment by SharpNoiZy** [comment]
No, not for me, everything seems to be fine until now.

**Comment by aL3891** [comment]
Hello! is there any issue tracking a protocol V3 server? i was a litte curious about writing an asp.net 5 compatible one but the spec on the wiki seems a little outdated, perhaps im just not looking in the right place?

**Comment by maartenba** [comment]
https://github.com/NuGet/NuGetGallery/issues/2632

Do know we're starting work on this soon.

**Comment by aL3891** [comment]
If you publish the spec i might be able to help :)

**Comment by maartenba** [comment]
True :-) Keep an eye on our GitHub...

**Comment by wpostma** [comment]
So due to this being open still, does that mean it is still not done?

**Comment by yishaigalatzer** [comment]
Yes it is not being implemented. Note that the V3 protocol is not required by any client and is purely optional.

Although it does provide some neat performance improvements, we are not able to prioritize this work high enough compared to other hot issues on the list.

**Comment by wpostma** [comment]
Ok. API V2 is working great.

**Comment by yishaigalatzer** [comment]
Wonderful, thanks for confirming.

**Comment by mkeymolen** [comment]
dotnet restore does't work with my private nuget server (setup with the Nuget.Server 2.11.2 package) 
`error: Unable to load the service index for source http://myhostname/nuget.
error:   The content at 'http://myhostname/nuget' is not a valid JSON object.
error:   Unexpected character encountered while parsing value: <. Path '', line 0, position 0.`

anyone an idea how to fix this?

**Comment by yishaigalatzer** [comment]
Does nuget.exe restore work for you? It should produce the exact same results.

**Comment by wpostma** [comment]
It looks like dotnet restore is being pointed at a URL which is not a nuget feed.  Maybe your URL is missing something or has something extra added.  I would poke around with your browser and experiment until you find the actual URL that works with nuget.  The .net application is mounted at http://myhostname/nuget and the NUGET Feed name might be something inside it like  http://myhostname/nuget/nuget/ or so.

If you open your browser at http://myhostname/nuget/nuget/Packages and that works, then try  http://myhostname/nuget/nuget as your feed URL.

**Comment by mkeymolen** [comment]
thanks for the feedback. i found the solution. @wpostma you pointed me in the right direction, thanks for that.
All was working fine on my windows machine but not on my Mac, so i my feed was not the problem. I took a second look to my nuget.config file on my mac and than i saw a stupid mistake. Protocol version of my private package source was defined on 3. I changed it to 2 because the nuget.server package is still 2 and problem was solved. Thanks for your help!
